crypto-core/src/main/java/com/palantir/crypto/cipher/AesCtrCipher.java

```diff
@@ -89,21 +86,20 @@ public Cipher seek(long pos) {
             newIv = new IvParameterSpec(tmpIv);
         }
 
-        Cipher cipher = getInstance();
-
-        // Init the cipher with the new iv
         try {
+            CryptoCipher cipher = CryptoCipherFactory.getCryptoCipher(ALGORITHM);
             cipher.init(currentOpmode, key, newIv);
-        } catch (InvalidKeyException | InvalidAlgorithmParameterException e) {
+
+            // Skip to the byte offset in the block where 'pos' is located
+            int bytesToSkip = (int) (pos % BLOCK_SIZE);
+            byte[] skip = new byte[bytesToSkip];
+            byte[] out = new byte[BLOCK_SIZE];
+            cipher.update(skip, 0, bytesToSkip, out, 0);
+            return cipher;
+        } catch (GeneralSecurityException e) {
```

Comment:
Reason for the `GeneralSecurityException` is that it's what `CryptoCipherFactory.getCryptoCipher()` throws, and then the exceptions that the other methods in the try block throw are all subclasses of that.